tests/functional/manila_controller_test.go

```diff
@@ -0,0 +1,387 @@
+			DeferCleanup(th.DeleteKeystoneAPI, th.CreateKeystoneAPI(namespace))
+		})
+		It("Should set DBReady Condition and set DatabaseHostname Status when DB is Created", func() {
+			th.SimulateMariaDBDatabaseCompleted(manilaTest.Instance)
+			th.SimulateJobSuccess(manilaTest.ManilaDBSync)
+			Manila := GetManila(manilaTest.Instance)
+			Expect(Manila.Status.DatabaseHostname).To(Equal("hostname-for-openstack"))
+			th.ExpectCondition(
+				manilaName,
+				ConditionGetterFunc(ManilaConditionGetter),
+				condition.DBReadyCondition,
+				corev1.ConditionTrue,
+			)
+			th.ExpectCondition(
+				manilaName,
+				ConditionGetterFunc(ManilaConditionGetter),
+				condition.DBSyncReadyCondition,
+				corev1.ConditionFalse,
+			)
+		})
+		It("Should fail if db-sync job fails when DB is Created", func() {
+			th.SimulateMariaDBDatabaseCompleted(manilaTest.Instance)
+			th.SimulateJobFailure(manilaTest.ManilaDBSync)
+			th.ExpectCondition(
+				manilaTest.Instance,
+				ConditionGetterFunc(ManilaConditionGetter),
+				condition.DBReadyCondition,
+				corev1.ConditionTrue,
+			)
+			th.ExpectCondition(
+				manilaTest.Instance,
+				ConditionGetterFunc(ManilaConditionGetter),
+				condition.DBSyncReadyCondition,
+				corev1.ConditionFalse,
+			)
+		})
+		It("Does not create ManilaAPI", func() {
+			ManilaAPINotExists(manilaTest.Instance)
+		})
+		It("Does not create ManilaScheduler", func() {
+			ManilaSchedulerNotExists(manilaTest.Instance)
+		})
+		It("Does not create ManilaShare", func() {
+			ManilaShareNotExists(manilaTest.Instance)
+		})
+	})
+	When("Both TransportURL secret and osp-secret are available", func() {
+		BeforeEach(func() {
+			DeferCleanup(th.DeleteInstance, CreateManila(manilaTest.Instance, GetDefaultManilaSpec()))
+			DeferCleanup(k8sClient.Delete, ctx, CreateManilaMessageBusSecret(manilaTest.Instance.Namespace, manilaTest.RabbitmqSecretName))
+			DeferCleanup(th.DeleteDBService, th.CreateDBService(
+				manilaTest.Instance.Namespace,
+				GetManila(manilaName).Spec.DatabaseInstance,
+				corev1.ServiceSpec{
+					Ports: []corev1.ServicePort{{Port: 3306}},
+				},
+			),
+			)
+			th.SimulateTransportURLReady(manilaTest.ManilaTransportURL)
+		})
+		It("should create config-data and scripts ConfigMaps", func() {
+			keystoneAPI := th.CreateKeystoneAPI(manilaTest.Instance.Namespace)
+			DeferCleanup(th.DeleteKeystoneAPI, keystoneAPI)
+			Eventually(func() corev1.ConfigMap {
+				return *th.GetConfigMap(manilaTest.ManilaConfigMapData)
+			}, timeout, interval).ShouldNot(BeNil())
+			Eventually(func() corev1.ConfigMap {
+				return *th.GetConfigMap(manilaTest.ManilaConfigMapScripts)
+			}, timeout, interval).ShouldNot(BeNil())
+		})
+	})
+	When("Manila CR is created without container images defined", func() {
+		BeforeEach(func() {
+			DeferCleanup(th.DeleteInstance, CreateManila(manilaTest.Instance, GetDefaultManilaSpec()))
+		})
+		It("has the expected container image defaults", func() {
```

Comment:
The current default Specs in base_test contains the container images so this is not really testing any defaulting at the moment. But as I noted there you can enable the webhook in suit_test.go and then drop the images from the specs.

Reply:
Yeah, which is addressed here:  https://github.com/openstack-k8s-operators/manila-operator/pull/77